lib/rbac/filterer.rb

```diff
@@ -424,6 +424,10 @@ def scope_to_tenant(scope, user, miq_group)
       user_or_group = user || miq_group
       tenant_id_clause = klass.tenant_id_clause(user_or_group)
 
+      if klass.respond_to?(:tenant_joins_clause)
```

Comment:
@rwsu I was thinking about it like this ^^, your code nicely adding cloud tenant scoping to the tenant scoping.

but I think that should be exclusive  because 
why we need cloud tenant scoping ? it is because we cannot use classic tenant scoping on cloud objects because there is missing relation to tenant.
for this case
there is tenant mapping and relation is established thru cloud_class -> cloud_tenant -> tenant
as you know.

So this is the reason why it have to be exclusive.

When in future(this is not likely) will cloud objects get the relation directly to tenant, then there will be used classic  tenancy mixin.

and my patch as you said yes some duplicated code but I believe we can clean up later, (maybe it can be moved to tenancy mixins )
there is important to see structure scoping and what why it is happening in rbac.